src/lib/color.ts

```diff
@@ -1,53 +1,67 @@
 import {
-  convertOklabToRgb,
-  convertOkhsvToOklab,
-  convertRgbToOklab,
-  convertOklabToOkhsv,
-  convertOkhslToOklab,
+	convertOklabToRgb,
+	convertOkhsvToOklab,
+	convertRgbToOklab,
+	convertOklabToOkhsv,
+	convertOkhslToOklab,
 } from 'culori/fn'
 
-
 export interface RGB {
-  r: number
-  g: number
-  b: number
+	r: number
+	g: number
+	b: number
 }
 
 export interface OKlab {
-  l: number
-  a: number
-  b: number
+	l: number
+	a: number
+	b: number
 }
 
 export interface OKhsv {
-  h: number
-  s: number
-  v: number
+	h: number
+	s: number
+	v: number
 }
 
 export interface OKhsl {
-  h: number
-  s: number
-  l: number
+	h: number
+	s: number
+	l: number
 }
 
 export function oklab_to_rgb(oklab: OKlab): RGB {
-  return convertOklabToRgb(oklab)
+	return convertOklabToRgb(oklab)
 }
 
 export function oklab_to_okhsv(oklab: OKlab): OKhsv {
-  const { h, s, v } = convertOklabToOkhsv(oklab)
-  return { h: h ?? 0, s, v }
+	const { h, s, v } = convertOklabToOkhsv(oklab)
+	return { h: h ?? 0, s, v }
 }
 
 export function okhsv_to_oklab(okhsv: OKhsv): OKlab {
-  return convertOkhsvToOklab(okhsv)
+	return convertOkhsvToOklab(okhsv)
 }
 
 export function rgb_to_oklab(rgb: RGB): OKlab {
-  return convertRgbToOklab(rgb)
+	return convertRgbToOklab(rgb)
 }
 
 export function okhsl_to_oklab(okhsl: OKhsl): OKlab {
-  return convertOkhslToOklab(okhsl)
-}
\ No newline at end of file
+	return convertOkhslToOklab(okhsl)
+}
+
+export function clamp_color(okhsv: OKhsv): OKhsv {
+	// only clamp if necessary
+	if (okhsv.s > 1 || okhsv.v > 1) {
+		const { r, g, b } = oklab_to_rgb(okhsv_to_oklab(okhsv))
```

Comment:
This seems a bit roundabout—converting to RGB just to clamp—but the problem isn’t really fixed otherwise. And I made it as performant as I could be by not clamping unless it’s actually needed (i.e. you can call `clamp_color()` for “free” on in-range colors)

Reply:
Also, edited the PR notes, but if we wanted to do this the “right” way, it’d just involve using Culori’s `clampChroma(…, 'oklch')` method.

This is the same method used on [oklch.com](https://oklch.com/#70,0.3,287,100) for the “fallback” color which matches Björn’s implementation from my extensive testing